docs/architecture/adr-042-state-sync.md

```diff
@@ -0,0 +1,209 @@
+
+However, upon analysis of the approaches with the orthogonal axis of
+eager/lazy and chunk/snapshot validation suggest that a version of
+WarpSync with per chunk light client validation might get the optimal
+performance and safety.
+
+## Decision: WarpSync With Per Chunk Light Client Validation
+
+The conclusion after thorough concideration of the
+advantages/disadvances of eager/lazy is to produce a state sync which
+replicates the performance IO characteristic of WarpSync and the security of
+per chunk validation. Manifests would be queries from nodes but be
+static across peers, providing a mapping state to pre-computed chunks
+allowing the snapshot to be downloaded in parallel.
+
+### Implementation
+
+* A new StateSync reactor brokers message transmission between the peers
+  and the ABCI application
+* A set of ABCI Messages
+* Design SnapshotFormat as an interface which can:
+    * validate chunks
+    * read/write chunks from file
+    * read/write chunks to/from application state store
+    * convert manifests into chunkRequest ABCI messages
+* Implement SnapshotFormat for cosmos-hub with concrete implementation for:
+    * read/write chunks in a way which can be:
+        * parallelized accros peers
+        * validated on receipt
+    * read/write to/from IAVL+ tree
+
+![StateSync Architecture Diagram](img/state-sync.png)
+
+## Implementation Path
+
+* Create StateSync reactor based on  [#3753](https://github.com/tendermint/tendermint/pull/3753)
+* Design SnapshotFormat with an eye towards cosmos-hub implementation
+* ABCI message to send/receive SnapshotFormat
+* IAVL+ changes to support SnapshotFormat
+* Deliver Warp sync (no chunk validation)
+* light client implementation for weak subjectivity
+* Deliver StateSync with chunk validation
+
+## Status
+
+Proposed
+
+## Concequences
+
+### Neutral
+
+### Positive
+* Safe & performant state sync design substantiated with real world implementation experience
+* General interfaces allowing application specific innovation
+* Parallizable implementation trajectory with reasonable engineering effort
+
+### Negative
+* Static Scheduling lacks opportunity for real time chunk availability optimizations
+
+## References
+[sync: Sync current state without full replay for Applications](https://github.com/tendermint/tendermint/issues/828) - original issue
+[tendermint state sync proposal](https://docs.google.com/document/d/15MFsQtNA0MGBv7F096FFWRDzQ1vR6_dics5Y49vF8JU/edit?ts=5a0f3629) - Cloudhead proposal
+[tendermint state sync proposal 2](https://docs.google.com/document/d/1npGTAa1qxe8EQZ1wG0a0Sip9t5oX2vYZNUDwr_LVRR4/edit) - ackratos proposal
+[proposal 2 implementation](https://github.com/tendermint/tendermint/pull/3243)  - ackratos implementation
+[WIP General/Lazy State-Sync pseudo-spec](https://github.com/tendermint/tendermint/issues/3639) - Jae Proposal
+[Warp Sync Implementation](https://github.com/tendermint/tendermint/pull/3594) - ackratos
```

Comment:
we have some more stuff:
1. document for this enhancement: https://github.com/binance-chain/BEPs/pull/18, included more chunking strategy we adopted
2. to try out state sync in binance chain prod and testnet: https://github.com/binance-chain/docs-site/blob/master/docs/fullnode.md#state-sync